Rearrange config objects

ttung opened the pull request:
### Description
Move most config objects to the top level.  Maintain two second-level config objects:

1. One for holding and caching AWS secrets.  This is separate because we don't always want to trigger AWS secrets at app startup.
2. `DatabaseConfig`, which remains separate because not all fields are defined for every valid configuration.  Flask will retrieve all fields.

Depends on #104 

### Test plan
Deployed to Elastic Beanstalk, and verified I can log in.

phoenixAja commented on src/py/aspen/config/config.py:
```diff
@@ -24,94 +26,110 @@ def __init_subclass__(cls: Type[Config], descriptive_name: str, *args, **kwargs)
     @property
-    def DEBUG(self):
-        return False
+    @abc.abstractmethod
```
this is much better

phoenixAja commented on src/py/aspen/config/production.py:
```diff
@@ -1,8 +1,29 @@
+import uuid
+from typing import Any, Mapping
+
+from .config import Config, DatabaseConfig, SecretsConfig
```
should we change to aspen.config?

ttung replied:
ok you seem to really dislike relative imports, i will dump them all in a fire in a single PR coming up. :)

phoenixAja replied:
no no no i was just thinking because now we have a smattering of relative imports and aspen imports. But a unifying PR _would_ be nice 😉

ttung replied:
believe it or not, there is actually a pattern and a rationale to when i prefer relative and absolute.

Commit:
Rearrange config objects

### Description
Move most config objects to the top level.  Maintain two second-level config objects:

1. One for holding and caching AWS secrets.  This is separate because we don't always want to trigger AWS secrets at app startup.
2. `DatabaseConfig`, which remains separate because not all fields are defined for every valid configuration.  Flask will retrieve all fields.

### Test plan
Deployed to Elastic Beanstalk, and verified I can log in.
